api/handlers/container/create.go

```diff
@@ -187,6 +187,10 @@ func (h *handler) create(w http.ResponseWriter, r *http.Request) {
 	if req.HostConfig.SecurityOpt != nil {
 		securityOpt = req.HostConfig.SecurityOpt
 	}
+	cgroupnsMode := defaults.CgroupnsMode()
+	if req.HostConfig.CgroupnsMode.Valid() {
```

Comment:
qq: if the requested cgroup mode is not valid, shouldn't we just through an error instead of setting the default. Side note, we should at least have a log here that we are ignoring the cgroup mode. What is dockers behavior?

Reply:
docker simply rejects the request and throws an error - https://github.com/moby/moby/blob/0e2cc22d36ae3013f83863c8da2e1b808f25e78e/daemon/oci_linux.go#L349

You're right, we're effectively filtering out bad requests without letting the user know that the option sent is wrong.

Will update this